### ISSUE TITLE
Support for float datatypes

### ISSUE THREAD

**Issue by shauryapatel1995** [NVIDIA/nvcomp#14]
Hi,
Is there any plan to support float datatypes with standard algorithms in the future? 
I think currently support is restricted to int datatypes?

**Comment by nv-dlasalle** [comment]
The best way to compress floating point data currently, is to use the LZ4 compressor, which operates at the byte level anyways:
```
float * float_data = ...;

nvcomp::LZ4Compressor(reinterpret_cast<uint8_t*>(float_data), size*sizeof(float), 65536);
...
```

It is unlikely Cascaded will provide much compression given how floating point numbers are represented, but it is still doable by casting to an equal width integer type:
```
float * float_data = ...;

// only applying RLE, assuming delta and bit-packing of the floating point data won't result in compression
nvcomp::LZ4Compressor(reinterpret_cast<uint32_t*>(float_data), size, 1, 0, 0);
...
```

**Comment by shauryapatel1995** [comment]
Makes sense, but can we add support for float data type natively anyway to not have to use this workaround, I am willing to contribute that if you guys are okay with it.

**Comment by postsql** [comment]
In https://blog.timescale.com/blog/time-series-compression-algorithms-explained/#floating-point-compression
they claim good floating-point compression performance via XOR-ing

**Comment by nsakharnykh** [comment]
@postsql this is an interesting idea, thanks for sharing the link. I would consider XOR as a variant of Delta compression. It can help with floating numbers, but also may work well for some integer sequences. I suppose we can design a way to add it to the cascaded scheme.

**Comment by felipeblazing** [comment]
Another way of compressing floating point data is using dictinoary encoding when the cardinality is low. If you dictionary encode the columns and the number of unique values is less than 65k  you can then bit pack the indices into the dictionary.
